server/src/main.cc

```diff
@@ -1,45 +1,54 @@
-#include <oatpp/core/macro/component.hpp>
-#include <oatpp/core/provider/Provider.hpp>
-#include <oatpp/network/ConnectionHandler.hpp>
-#include <oatpp/network/ConnectionProvider.hpp>
-#include <oatpp/network/Server.hpp>
-#include <oatpp/web/server/HttpRouter.hpp>
 #include <oatpp/web/server/api/Endpoint.hpp>
+#include <oatpp/network/Server.hpp>
+#include <oatpp-sqlite/Connection.hpp>
+#include <oatpp/core/macro/component.hpp>
 #include <oatpp-swagger/Controller.hpp>
 
-#include "component/app_component.hh"
-#include "controller/user_controller.hh"
-#include "controller/offer_controller.hh"
+#include "component/AppComponent.hh"
+#include "controller/AuthController.hh"
+#include "controller/OfferController.hh"
+#include "controller/UserController.hh"
+#include "controller/PlaceController.hh"
 
 int
-main()
-{
-    ::oatpp::base::Environment::init();
+main() {
+    oatpp::base::Environment::init();
 
-    ::server::component::app_component const app_component{};
+    server::component::AppComponent appComponent;
 
-    OATPP_COMPONENT(::std::shared_ptr<::oatpp::web::server::HttpRouter>, router);
+    auto router = appComponent.httpRouterComponent.getObject();
 
-    ::oatpp::web::server::api::Endpoints endpoints{};
-    endpoints.append(router->addController(::server::controller::user_controller::create_shared())->getEndpoints());
-    endpoints.append(router->addController(::server::controller::offer_controller::create_shared())->getEndpoints());
+    oatpp::web::server::api::Endpoints endpoints;
 
-    router->addController(::oatpp::swagger::Controller::createShared(endpoints));
+    endpoints.append(router->addController(server::controller::AuthController::create_shared())->getEndpoints());
+    endpoints.append(router->addController(server::controller::OfferController::createShared())->getEndpoints());
+    endpoints.append(router->addController(server::controller::PlaceController::createShared())->getEndpoints());
+    endpoints.append(router->addController(server::controller::UserController::createShared())->getEndpoints());
 
-    OATPP_COMPONENT(::std::shared_ptr<::oatpp::network::ServerConnectionProvider>, server_connection_provider);
-    OATPP_COMPONENT(::std::shared_ptr<::oatpp::network::ConnectionHandler>, connection_handler);
 
-    ::oatpp::network::Server server{server_connection_provider, connection_handler};
-    OATPP_LOGD("Server", "Server running on port %s", server_connection_provider->getProperty("port").toString()->c_str());
+    router->addController(oatpp::swagger::Controller::createShared(endpoints));
+
+    oatpp::network::Server server(
+            appComponent.serverConnectionProviderComponent.getObject(),
+            appComponent.serverConnectionHandlerComponent.getObject()
+    );
+
+    auto const host = appComponent.serverConnectionProviderComponent.getObject()->getProperty("host").toString();
+    auto const address = "http://" + (host->find(":") == std::string::npos ? host : "[" + host + "]");
```

Comment:
Could you make a comment on what is happening here?

Reply:
If it's not readable you should see the modified file only (without changes). I just edited the file to include all controllers and changed their names so git doesn't properly recognize the changes. I also removed the `::` prefix everywhere I could.

L37 and L38 generates the URL address of the server for the log. L38 adds the square braces if the host address is ipv6.

You may want to see [my comment above](https://github.com/mnurzyns/ParkSpaceSharing/pull/36/files/eaf878e5106238861b3c7c1f7a74ae917f92e110#r1179564371)

Reply:
I'm talking about this line 38:
```
auto const address = "http://" + (host->find(":") == std::string::npos ? host : "[" + host + "]");
```
Why are you wrapping the host address in square brackets when it contains a colon?

Reply:
> L38 adds the square braces if the host address is ipv6.

Yes, that is what I meant. I didn't know about the square-bracket-wrapping-for-ipv6 thingy.

Reply:
Maybe we should add a comment for that? Just to mention it's for IPv6, because I don't thing everybody knows about it either.